config/passport.js

```diff
@@ -1,28 +1,65 @@
-import passport from "passport";
-import strategies from "passport-local";
-import models from '../models/';
+import passport from 'passport';
+import strategies from 'passport-local';
+import { Strategy as JWTStrategy, ExtractJwt } from 'passport-jwt';
+import dotenv from 'dotenv';
+import models from '../models';
+
+dotenv.config();
 
 const LocalStrategy = strategies.Strategy;
-const User = models.User;
+const { User } = models;
 
-passport.use(
+passport.use('signup',
   new LocalStrategy(
     {
-      usernameField: "user[email]",
-      passwordField: "user[password]"
+      usernameField: User.email,
+      passwordField: User.password
     },
-    function(email, password, done) {
-      User.findOne({ email: email })
-        .then(function(user) {
-          if (!user || !user.validPassword(password)) {
-            return done(null, false, {
-              errors: { "email or password": "is invalid" }
-            });
-          }
+    async (email, password, done) => {
+      try {
+        const user = await User.findOne({ where: { email: email } });
+        if(!email || !password) {
```

Comment:
I think here you are validating email and password after using them. You used email to find a user and after that, you check if the email is not provided! I suggest you should validate the parameters before doing the query like this:
```javascript
if(!email || !password) {...return the error here}
const user = await User.findOne({ where: { email: email } });
```